canopy_driver.F90

```diff
@@ -210,13 +211,32 @@ program canopy_driver
                     z0ghcm, cdrag, pai, canBOT(i), canTOP(i), canWIND(i, loc))
             end do
 
-! ... calculate wind adjustment factor dependent on canopy winds and fire type
-
+! ... calculate wind adjustment factor dependent on fires (FRP), canopy winds,
+!                                                        flameh, and fire type
+            if (flameh_opt .eq. 0) then
+                if (nlon .eq. 1) then      !single lon point -- reset to user value
+                    flameh = flameh_set
+                else                       !calculate flameh
+                    dx     = dlon*111000.0  !convert lon grid to distance (m)
```

Comment:
OK, we can switch to latitude maybe later, as the input_variables file has longitude always varying.  I agree to move dx up, and take absolute to avoid possible negatives. Better?

Reply:
Yes, it is more robust now. What I was alluding to though is that the value of `dx` in such a lat/lon grid changes with latitude. And we could account for that. But maybe that can wait until we set up to work with 2-d array inputs (e.g. loaded from netCDF files).